common/src/lib.rs

```diff
@@ -2236,13 +2236,28 @@ impl JsonSchema for Node {
 #[cfg_attr(feature = "serde", serde(deny_unknown_fields))]
 #[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
 pub struct Tree {
+    /// The identifier of the tree's root node
     pub root: NodeId,
+    /// The name of the application this tree belongs to.
+    /// This value is optional.
+    pub app_name: Option<String>,
```

Comment:
Thinking more about this, I wonder if we shouldn't require the app name afterall. Some UI toolkits/game engines force users to provide one to manage config files and stuff, and without the app name ATs wouldn't have a reliable way to patch an app.

Just a thought, feel free to disagree.

Reply:
I'm unopinionated about this. I'm also not sure how we'd use the app_name on macOS yet.